src/CommunityToolkit/Xamarin.CommunityToolkit/Extensions/Navigation/NavigationExtensions.uwp.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using Xamarin.CommunityToolkit.UI.Views;
+using Xamarin.Forms;
+using Xamarin.Forms.Platform.UWP;
+
+namespace Xamarin.CommunityToolkit.Extensions
+{
+	public static partial class NavigationExtensions
+	{
+		static void OnShowPopup(BasePopup popup)
+		{
+			popup.Parent = GetCurrentPage(Application.Current.MainPage);
+			Platform.CreateRenderer(popup);
+
+			// https://github.com/xamarin/Xamarin.Forms/blob/0c95d0976cc089fe72476fb037851a64987de83c/Xamarin.Forms.Platform.iOS/PageExtensions.cs#L44
+			Page GetCurrentPage(Page currentPage)
```

Comment:
I decided to place it here since it isn't used anywhere else in XCT. This is a very useful recursive function and I could see value in moving it as an **internal** extension method.

Do you think it should be public or internal?

Reply:
Let's make it internal for now. If it's public we can't break it xP